MAINT: signal: build warning (`sprintf`) on macOS

To close this issue, you should replace the use of `sprintf` in `scipy/signal/_sigtoolsmodule.cc` with `snprintf` as per the warning message.

```
[1311/1464] Compiling C++ object scipy/signal/_sigtools.cpython-311-darwin.so.p/_sigtoolsmodule.cc.o
../scipy/signal/_sigtoolsmodule.cc:910:13: warning: 'sprintf' is deprecated: This function is provided for compatibility reasons only.  Due to security concerns inherent in the design of sprintf(3), it is highly recommended that you use snprintf(3) instead. [-Wdeprecated-declarations]
  910 |             sprintf(mystr, "Failure to converge at iteration %d, try reducing transition band width.\n", niter);
      |             ^
/Library/Developer/CommandLineTools/SDKs/MacOSX14.0.sdk/usr/include/stdio.h:180:1: note: 'sprintf' has been explicitly marked deprecated here
  180 | __deprecated_msg("This function is provided for compatibility reasons only.  Due to security concerns inherent in the design of sprintf(3), it is highly recommended that you use snprintf(3) instead.")
      | ^
/Library/Developer/CommandLineTools/SDKs/MacOSX14.0.sdk/usr/include/sys/cdefs.h:215:48: note: expanded from macro '__deprecated_msg'
  215 |         #define __deprecated_msg(_msg) __attribute__((__deprecated__(_msg)))
      |                                                       ^
1 warning generated.
```

--- a/scipy/signal/_sigtoolsmodule.cc
+++ b/scipy/signal/_sigtoolsmodule.cc
@@ -907,7 +907,7 @@ static PyObject *_sigtools_remez(PyObject *NPY_UNUSED(dummy), PyObject *args)
                     type, maxiter, grid_density, &niter);
     if (err < 0) {
         if (err == -1) {
-            sprintf(mystr, "Failure to converge at iteration %d, try reducing transition band width.\n", niter);
+            snprintf(mystr, sizeof(mystr), "Failure to converge at iteration %d, try reducing transition band width.\n", niter);
 	        PyErr_SetString(PyExc_ValueError, mystr);
 	        goto fail;
         }

